### DISCUSSION TITLE
The possibility of CommunityToolkit.Networking

### DISCUSSION THREAD

**Discussion by TheVeryStarlk** [CommunityToolkit/Labs-Windows#436] [Ideas]
The current APIs present in .NET for creating TCP clients/servers are outdated and old-school (I'm referring to the TcpClient/TcpServer classes). So, this leaves us with raw sockets, which is very error prone especially for those who are beginners in networking (Like me 😅), hence why I thought that creating a package that delivers APIs for networking in .NET that focuses on both high performance and ease of use would be a great idea.

And this does not just apply to the TCP part of things, but the same thing also applies to UDP, WS, etc...

**Comment by kmgallahan** [comment]
Wrong repo: https://github.com/CommunityToolkit/dotnet

> are outdated and old-school

It would probably be best to identify specific shortfalls and comparisons to better implementations to get more traction.

**Comment by TheVeryStarlk** [comment]
Oh. I'm so sorry, I was directed to send it here. I have [created a new one](https://github.com/CommunityToolkit/dotnet/discussions/707) and modified it to show what I meant by outdated and old-school.
Should I delete this discussion?

**Reply by kmgallahan** [reply]
> Should I delete this discussion?

I'm not sure if non-maintainers can delete or close discussions.

**Comment by TheVeryStarlk** [comment]
I can close it, should I close it as duplicate?

**Reply by kmgallahan** [reply]
Probably best to only keep open the discussion in the .NET repo as this one is for Windows-specific stuff.
